pulser-core/pulser/parametrized/variable.py

```diff
@@ -71,9 +73,10 @@ def _assign(self, value: Union[ArrayLike, float, int]) -> None:
         object.__setattr__(self, "_count", self._count + 1)
 
     def _validate_value(
-        self, value: Union[ArrayLike, float, int]
-    ) -> np.ndarray:
-        val = np.array(value, dtype=self.dtype, ndmin=1)
+        self, value: Union[AbstractArrayLike, float, int]
+    ) -> pm.AbstractArray:
+        val = pm.AbstractArray(value)
+        # val = np.array(value, dtype=self.dtype, ndmin=1)
```

Comment:
The casting to `self.dtype` and the `ndmin=1` are both important. `AbstractArray` already has a `dtype` argument, we should add `ndmin` too

Reply:
Ok, I can do that. But what's the actual reason for this?

Reply:
A variable's value is always a 1D array, even when it has only one entry. We accept 0D values but then internally turn them into a single-entry 1D array through this `ndmin=1` option

Reply:
yes, makes sense

Reply:
What if we skip this argument completely and simply make sure that any tensor/array gets reshaped to 1D shape if it is 0D? I noticed that `ndmin` is used only in `variable.py` and with value 1, so I think there's no need to fully replicate the behavior of `ndmin` of numpy here.

Reply:
> What if we skip this argument completely and simply make sure that any tensor/array gets reshaped to 1D shape if it is 0D?

I think that is a strong and potentially problematic assumption that I would prefer to avoid making in general.

> I noticed that `ndmin` is used only in `variable.py` and with value 1, so I think there's no need to fully replicate the behavior of `ndmin` of numpy here.

I agree that there's no need to fully replicate the behaviour of `ndmin`. You can make a restricted `force_array: bool` argument for this purpose, how about that?